### ISSUE TITLE
Custom drop target

### ISSUE THREAD

**Issue by artooras** [clauderic/react-sortable-hoc#360]
I understand that this library is intended purely for sorting lists, for which it works great. However, I need not only sort a list, but also drag items from the list onto a custom drop target outside of the list. Can I achieve that with react-sortable-hoc? Is it perhaps compatible with other dnd libraries?

**Comment by CyrilQuandalle** [comment]
No, I meant a list with react-sortable-hoc with each element of the list wrapped in a component implementing react-dnd dragSource

**Comment by artooras** [comment]
I tried doing that some time ago, but react-sortable-hoc list sorting stopped working for me. Hence my question. Do you have a sample of code to share by any chance?

**Comment by CyrilQuandalle** [comment]
You will find bellow a component wrapper implementing react-dnd dragSource, try to wrap each item of your SortableHoc.
`import React, { Component } from 'react'
import { DragSource } from 'react-dnd'
import PropTypes from 'prop-types'

const dragSource = {
  beginDrag (props) {
    return { payload: props.payload }
  }
}

function collect (connect) {
  return { connectDragSource: connect.dragSource() }
}

class DragContainer extends Component {
  render () {
    const { connectDragSource, children } = this.props
    return connectDragSource(
      <div className={ this.props.className }>
        { children }
      </div>
    )
  }
}

DragContainer.propTypes = {
  children: PropTypes.object,
  connectDragSource: PropTypes.func,
  className: PropTypes.string
}

DragContainer.displayName = 'DragContainer'

export default DragSource((props) => { return props.itemType }, dragSource, collect)(DragContainer)
`

**Comment by sbrichardson** [comment]
I've been doing a few customizations for multiple levels of drag targets, using this lib.
We can share info/notes if interested.

[Link to screen recording (youtube)](https://youtu.be/xBjis3SIHkY)
Here is a preview:

![builder-dev-preview-1](https://user-images.githubusercontent.com/8600215/40197130-c833a01e-59cf-11e8-9e96-9667a706d7a4.gif)

**Comment by fairps** [comment]
@sbrichardson that is really nice looking.  My personal journey on this stuff has been across multiple libaries looking for something that does exactly what I need, or that can be coerced :-) ... 
react-grid-layout: provides drag/drop reordering, different sized items, drag resizing of items... what it lacks is the ability to drag from one react-grid-layout into another.  There is a PR to support a drag API but it has been sitting unmerged forever
react-beautiful-dnd: really amazing for lists, but no "grid" concept and items have to be fixed size in one dimension (same height or same width depending on if it is a horizontal/vertical list or not)
And then there is this library...  have you managed to drag/drop between 2 instances of this control?  To this point I've just been poking around looking and haven't started trying to implement yet beyond what I have working in react-grid-layout.

**Comment by sbrichardson** [comment]
I'm actually combining some of that functionality as well. That preview is limited to that one component, but it's using grid-layout also, for the broader layout concept. What's an example use case you are trying to build for (if it's not page layout stuff)?

I'm using react-grid-layout also, but I'm currently simplifying it to output positioning based on flexbox vs the current exact positioning logic. Basically using the builder/drag interfaces to create declarative layouts, but converting that to web standards/flexbox etc where possible when the component is used on an actual page.
